[Android] Fails to build

This has been happening for a while, I think. This appears to be the relevant error:
```
In file included from ../../../../../src/mame/drivers/alpha68k.cpp:1:
In file included from ../../../../../src/emu/emu.h:104:
/home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:1258:12: error: 'consume' is a protected member of 'devcb_write<unsigned char, '\xFF'>::builder_base'
                { m_sink.consume(); }
                         ^
/home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:1497:11: note: in instantiation of function template specialization 'devcb_write<unsigned char, '\xFF'>::first_transform_builder<devcb_write<unsigned char, '\xFF'>::delegate_builder<device_delegate<void (address_space &, unsigned int, unsigned char, unsigned char)> >::wrapped_builder, (lambda at /home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:255:52)>::first_transform_builder<(lambda at /home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:255:52)>' requested here
                        return first_transform_builder<wrapped_builder, std::remove_reference_t<T> >(this->m_target, this->m_append, wrapped_builder(std::move(*this)), std::forward<T>(cb), this->exor(), in_mask, out_mask);
                               ^
/home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:255:42: note: in instantiation of function template specialization 'devcb_write<unsigned char, '\xFF'>::delegate_builder<device_delegate<void (address_space &, unsigned int, unsigned char, unsigned char)> >::transform<(lambda at /home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:255:52)>' requested here
                        auto trans(static_cast<Impl &>(*this).transform([val] (offs_t offset, T data, std::make_unsigned_t<T> &mem_mask) { mem_mask >>= val; return data >> val; }));
                                                              ^
/home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:1142:8: note: declared protected here
                void consume() { m_consumed = true; }
                     ^
/home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:1267:11: error: 'consume' is a protected member of 'devcb_write<unsigned char, '\xFF'>::builder_base'
                        m_sink.consume();
                               ^
/home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:255:9: note: in instantiation of member function 'devcb_write<unsigned char, '\xFF'>::first_transform_builder<devcb_write<unsigned char, '\xFF'>::delegate_builder<device_delegate<void (address_space &, unsigned int, unsigned char, unsigned char)> >::wrapped_builder, (lambda at /home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:255:52)>::first_transform_builder' requested here
                        auto trans(static_cast<Impl &>(*this).transform([val] (offs_t offset, T data, std::make_unsigned_t<T> &mem_mask) { mem_mask >>= val; return data >> val; }));
                             ^
/home/buildbot/buildbot/android/libretro-mame/build/projects/retro/mame/gmake-android-arm/../../../../../src/emu/devcb.h:1142:8: note: declared protected here
                void consume() { m_consumed = true; }
                     ^
2 errors generated.
alpha.make:627: recipe for target '../../../../libretro/obj/libretro/src/mame/drivers/alpha68k.o' failed
make[2]: *** [../../../../libretro/obj/libretro/src/mame/drivers/alpha68k.o] Error 1
Makefile:109: recipe for target 'alpha' failed
make[1]: *** [alpha] Error 2
make[1]: *** Waiting for unfinished jobs....
```
Rest of the log is here:
http://p.0bl.net/41004

http://paste.libretro.com/215233
It seems a problem in android SDK. Any android devs around?


`/home/buildbot/buildbot/android/libretro-mame/src/osd/libretro/retroprefix.h -o "../../../../libretro/obj/libretro/3rdparty/bgfx/3rdparty/dear-imgui/imgui.o" -c "../../../../../3rdparty/bgfx/3rdparty/dear-imgui/imgui.cpp"
In file included from ../../../../../3rdparty/bgfx/3rdparty/dear-imgui/imgui.cpp:770:
In file included from ../../../../../3rdparty/bgfx/3rdparty/dear-imgui/imgui.h:46:
In file included from /home/buildbot/tools/android/android-ndk-r20//sources/cxx-stl/llvm-libc++/include/string.h:61:
In file included from /home/buildbot/tools/android/android-ndk-r20//sysroot/usr/include/string.h:36:
In file included from /home/buildbot/tools/android/android-ndk-r20//sysroot/usr/include/bits/strcasecmp.h:37:
In file included from /home/buildbot/tools/android/android-ndk-r20//sysroot/usr/include/sys/types.h:39:
/home/buildbot/tools/android/android-ndk-r20//sysroot/usr/include/bits/pthread_types.h:38:3: error: unknown type name 'size_t'
  size_t stack_size;
  ^
/home/buildbot/tools/android/android-ndk-r20//sysroot/usr/include/bits/pthread_types.h:39:3: error: unknown type name 'size_t'
  size_t guard_size;
  ^
In file included from ../../../../../3rdparty/bgfx/3rdparty/dear-imgui/imgui.cpp:770:
In file included from ../../../../../3rdparty/bgfx/3rdparty/dear-imgui/imgui.h:46:
In file included from /home/buildbot/tools/android/android-ndk-r20//sources/cxx-stl/llvm-libc++/include/string.h:61:
In file included from /home/buildbot/tools/android/android-ndk-r20//sysroot/usr/include/string.h:36:
/home/buildbot/tools/android/android-ndk-r20//sysroot/usr/include/bits/strcasecmp.h:68:53: error: unknown type name 'size_t'; did you mean 'ssize_t'?`

--- a/scripts/toolchain.lua
+++ b/scripts/toolchain.lua
@@ -860,33 +860,14 @@ function toolchain(_buildDir, _subDir)
 
 	configuration { "android-*" }
 		objdir (_buildDir .. "android/obj/" .. _OPTIONS["PLATFORM"])
--- LIBRETRO HACK BEGIN support ndk-r13b structure
---		includedirs {
---			MAME_DIR .. "3rdparty/bgfx/3rdparty/khronos",
---			"$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/libcxx/include",
---			"$(ANDROID_NDK_ROOT)/sources/android/support/include",
---			"$(ANDROID_NDK_ROOT)/sources/android/native_app_glue",
---		}
-if os.getenv("ANDROID_NDK_ROOT") then		
-		checkndk13 = os.getenv("ANDROID_NDK_ROOT") .. "/sources/cxx-stl/llvm-libc++/libcxx/include/list"	
-		if (os.isfile(checkndk13)) then
-			includedirs {
-				"$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/libcxx/include",
-			}
-		else
-			includedirs {
-				"$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/include",
-			}
-		end
-end
 		includedirs {
 			MAME_DIR .. "3rdparty/bgfx/3rdparty/khronos",
-
-			"$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/libcxx/include",
-			"$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/include",
-			"$(ANDROID_NDK_ROOT)/sysroot/usr/include",
-			"$(ANDROID_NDK_ROOT)/sources/android/support/include",
-			"$(ANDROID_NDK_ROOT)/sources/android/native_app_glue",
+			--  LIBRETRO: don't mess with NDK includir order
+			-- "$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/libcxx/include",
+			-- "$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/include",
+			-- "$(ANDROID_NDK_ROOT)/sysroot/usr/include",
+			-- "$(ANDROID_NDK_ROOT)/sources/android/support/include",
+			-- "$(ANDROID_NDK_ROOT)/sources/android/native_app_glue",
 		}
 		linkoptions {
 			"-nostdlib",
@@ -894,49 +875,20 @@ end
 		flags {
 			"NoImportLib",
 		}
---		links {
---			"c",
---			"dl",
---			"m",
---			"android",
---			"log",
---			"c++_static",
---			"gcc",
---		}
 		links {
 			"c",
 			"dl",
 			"m",
 			"android",
 			"log",
-		}
-
-if os.getenv("ANDROID_NDK_ROOT") then
-		if (os.isfile(checkndk13)) then
-			links {
-				"c++_static",
-			}
-		else
-			links {
-				"c++_static",
-				"c++abi",
-				"unwind",
-				"android_support",
-			}
-		end
-end
-		links {
---=======
---			"c++_static",
---			"c++abi",
---			"android_support",
---			"stdc++",
--->>>>>>> 2beedc540f14267850036f8b2aba81e874895ade
+			"c++_static",
+			"c++abi",
 			"stdc++",
 			"gcc",
 		}
--- LIBRETRO HACK END support ndk-r13b structure
-
+		buildoptions_c {
+			"-Wno-strict-prototypes",
+		}
 		buildoptions {
 			"-fpic",
 			"-ffunction-sections",
@@ -960,14 +912,14 @@ end
 			"-Wl,-z,now",
 		}
 
-
 	configuration { "android-arm" }
 			libdirs {
 				"$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a",
 				"$(ANDROID_NDK_ROOT)/platforms/" .. androidPlatform .. "/arch-arm/usr/lib",
 			}
 			includedirs {
-				"$(ANDROID_NDK_ROOT)/sysroot/usr/include/arm-linux-androideabi",
+				--  LIBRETRO: don't mess with NDK includir order
+				-- "$(ANDROID_NDK_ROOT)/sysroot/usr/include/arm-linux-androideabi",
 			}
 			buildoptions {
 				"-gcc-toolchain $(ANDROID_NDK_ARM)",
@@ -981,7 +933,6 @@ end
 				"unwind",
 			}
 			linkoptions {
-				"$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a/libunwind.a -lgcc",
 				"-gcc-toolchain $(ANDROID_NDK_ARM)",
 				"--sysroot=$(ANDROID_NDK_ROOT)/platforms/" .. androidPlatform .. "/arch-arm",
 				"$(ANDROID_NDK_ROOT)/platforms/" .. androidPlatform .. "/arch-arm/usr/lib/crtbegin_so.o",

